stats/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package opentelemetry implements opencensus instrumentation code for gRPC-Go
+// clients and servers.
+package opentelemetry
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/internal"
+
+	"go.opentelemetry.io/otel/metric"
+)
+
+var logger = grpclog.Component("opentelemetry-instrumentation")
+
+var canonicalString = internal.CanonicalString.(func(codes.Code) string)
+
+var (
+	joinDialOptions = internal.JoinDialOptions.(func(...grpc.DialOption) grpc.DialOption)
+)
+
+// EmptyMetrics represents no metrics. To start from a clean slate if want to
+// pick a subset of metrics, use this and add onto it.
+var EmptyMetrics = Metrics{}
+
+// Metrics is a set of metrics to initialize. Once created, Metrics is
+// immutable.
```

Comment:
Switched to "to record". My logic of this was initialization and presence in setOfMetrics at init time determines whether metrics get recorded. I guess since A->B, and B->C, I can write that A does C (records).